sensing/preprocessor/gnss/gnss_poser/src/gnss_poser_core.cpp

```diff
@@ -13,76 +13,57 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include "gnss_poser/gnss_poser_core.h"
+#include "gnss_poser/gnss_poser_core.hpp"
+#include <rclcpp_components/register_node_macro.hpp>
 
 #include <algorithm>
 #include <string>
 
-#include <boost/circular_buffer.hpp>
-
-#include <ros/ros.h>
-
-#include <tf2/transform_datatypes.h>
-#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
-#include <tf2_ros/transform_broadcaster.h>
-
-#include <geometry_msgs/PoseStamped.h>
-#include <geometry_msgs/PoseWithCovarianceStamped.h>
-#include <sensor_msgs/NavSatFix.h>
-#include <std_msgs/Bool.h>
-#include <ublox_msgs/NavPVT.h>
-
-#include "gnss_poser/convert.h"
-
 namespace GNSSPoser
 {
-GNSSPoser::GNSSPoser(ros::NodeHandle nh, ros::NodeHandle private_nh)
-: nh_(nh),
-  private_nh_(private_nh),
+GNSSPoser::GNSSPoser(const rclcpp::NodeOptions & node_options)
+: rclcpp::Node("gnss_poser", node_options),
   tf2_listener_(tf2_buffer_),
-  coordinate_system_(CoordinateSystem::MGRS),
-  base_frame_("base_link"),
-  gnss_frame_("gnss"),
-  gnss_base_frame_("gnss_base_link"),
-  map_frame_("map"),
-  use_ublox_receiver_(false),
-  plane_zone_(9)
+  tf2_broadcaster_(this),
```

Comment:
Should this be `*this`?

Reply:
I couldn't find any references of people using `*this` in the ROS discourse though when doing some googling I did find that the geometry library tests were using `std::make_shared` and `shared_from_this()` so I changed the implementation to that:
* https://answers.ros.org/question/360516/ros2-tf2-broadcaster/ (implementation without `*this`)
* https://github.com/ros2/geometry2/blob/ros2/tf2_ros/test/test_transform_broadcaster.cpp#L46 (`shared_from_this()`)